### DISCUSSION TITLE
Question about the Unbiased Inverse Volume Rendering with Differential Trackers paper

### DISCUSSION THREAD

**Discussion by leroyvn** [mitsuba-renderer/mitsuba3#844] [Q&A] [answered]
I've been trying to use the local majorant implementation developed for the [Unbiased Inverse Volume Rendering with Differential Trackers](https://github.com/rgl-epfl/unbiased-inverse-volume-rendering) paper. I put together an [example relevant to my use case](https://gist.github.com/leroyvn/426287cbd2c81e408fb68a909683b51e) and observed artefacts when using the `volpath` integrator with a majorant grid. Are there issues or limitations I could have overlooked?

**Comment by merlinND** [comment]
Hello @leroyvn,

Have you checked the `volpathsimple` integrator [used in the paper](https://github.com/rgl-epfl/unbiased-inverse-volume-rendering/blob/master/python/integrators/volpathsimple.py) and the corresponding Mitsuba branch, e.g. [medium.cpp](https://github.com/mitsuba-renderer/mitsuba3/blob/unbiased-inverse-volume-rendering/src/render/medium.cpp)?
The [diff](https://github.com/mitsuba-renderer/mitsuba3/compare/master...unbiased-inverse-volume-rendering) is probably a good place to check for small details.

**Comment by leroyvn** [comment]
Hi @merlinND, thanks for your reply. I'm using the latest commit on the [`unbiased-inverse-volume-renderin`g](https://github.com/mitsuba-renderer/mitsuba3/tree/unbiased-inverse-volume-rendering) branch. I was aware that the `volpathsimple` integrator implements the methods presented in the paper, but I need surfaces in my scene. 

So maybe my question should rather be: were the results produced by the `volpath` with and without a majorant grid compared already? (Happy to investigate and fix issues, I just want to be sure that the artefacts I'm seeing are not a known issue already.)

**Reply by merlinND** [reply, marked as answer]
I don't think that the majorant grid was tested & validated for integrators other than the one used in the paper (the Python-based `volpathsimple` integrator).

**Reply by leroyvn** [reply]
Pretty clear, thanks!
